client/attest.go

```diff
@@ -36,6 +48,16 @@ type AttestOpts struct {
 	// Key.Attest() will construct the certificate chain by making GET requests to
 	// the contents of Key.cert.IssuingCertificateURL using this client.
 	CertChainFetcher *http.Client
+	// TEEDevice implements the TEEDevice interface for collecting a Trusted execution
+	// environment attestation. If nil, then Attest will try all known TEE devices,
+	// and TEENonce must be nil. If not nil, Attest will not call Close() on the device.
+	TEEDevice TEEDevice
```

Comment:
could you add a quick test that `Attest` fails when TEEDevice and TEENonce are xor set?

Reply:
Added.